src/gsy_e/gsy_e_core/user_profile_handler.py

```diff
@@ -267,7 +275,9 @@ def __init__(self):
     def activate(self):
         """Connect to DB, update current timestamp and get the first chunk of data from the DB"""
         self._connect_to_db()
-        self.update_time_and_buffer_profiles(GlobalConfig.start_date)
+        self._update_current_time(GlobalConfig.start_date)
+        if self.db:
```

Comment:
Is this `if self.db` check really needed? I'm asking this because `self._connect_to_db()` is already called and if DB is not accessible, it should raise an error probably. What do you think?